Core.ApplicationServices/Organizations/OrganizationUnitService.cs

```diff
@@ -0,0 +1,345 @@
+            if (unit.Failed)
+            {
+                return unit.Error;
+            }
+
+            if (!_authorizationContext.AllowDelete(unit.Value))
+            {
+                return new OperationError(OperationFailure.Forbidden);
+            }
+            
+            var result =  _organizationRightsService.RemoveUnitRightsByIds(organizationId, parameters.OrganizationUnitRights)
+                .Match
+                (
+                    error => error,
+                () => RemovePayments(parameters.PaymentRegistrationDetails)
+                )
+                .Match
+                (
+                    error => error,
+                    () => RemoveContractRegistrations(parameters.ItContractRegistrations)
+                )
+                .Match
+                (
+                    error => error,
+                    () => RemoveSystemResponsibleRegistrations(parameters.ResponsibleSystems)
+                )
+                .Match
+                (
+                    error => error,
+                    () => RemoveSystemRelevantUnits(parameters.RelevantSystems, unitId)
+                );
+
+            if (result.HasValue)
+            {
+                return result.Value;
+            }
+
+            transaction.Commit();
+            return Maybe<OperationError>.None;
+        }
+
+        public Maybe<OperationError> DeleteAllUnitOrganizationRegistrations(int organizationId, int unitId)
+        {
+            using var transaction = _transactionManager.Begin();
+
+            var canModifyOrganization = CanModifyOrganization(organizationId);
+            if (canModifyOrganization.HasValue)
+            {
+                return canModifyOrganization.Value;
+            }
+
+            var deleteRegistrationsResult = GetOrganizationRegistrations(organizationId, unitId)
+                .Match(val => DeleteSelectedOrganizationRegistrations(organizationId, unitId, ToChangeParametersFromRegistrationDetails(val)),
+                    error => error);
+
+            if (deleteRegistrationsResult.HasValue)
+                return deleteRegistrationsResult.Value;
+            
+            transaction.Commit();
+
+            return Maybe<OperationError>.None;
+        }
+
+        public Maybe<OperationError> TransferSelectedOrganizationRegistrations(int organizationId, int unitId, int targetUnitId, OrganizationRegistrationChangeParameters parameters)
+        {
+            using var transaction = _transactionManager.Begin();
+
+            var canModifyOrganization = CanModifyOrganization(organizationId);
+            if (canModifyOrganization.HasValue)
+            {
+                return canModifyOrganization.Value;
+            }
+
+            var unit = GetOrganziationUnit(unitId);
+            if (unit.Failed)
+            {
+                return unit.Error;
+            }
+
+            var targetUnit = GetOrganziationUnit(targetUnitId);
```

Comment:
Again organization units are not fetched from the org.. just by id from the giant db